### PR TITLE
Fix boundary id mapping

### PR DESCRIPTION
**Pull request by wdhawkins**
This PR fixes an issue with boundary id mapping in lbs_solver. The global id mapping was changed in #255, but a routine in lbs_solver was still using hardcoded values corresponding to the pre-#255 mapping. In fixing this issue, I've tried to bring some consistency to the boundary id mapping (even in places where it's not technically necessary). 

With this fix, all of the regression inputs that were modified in #255 needed to be reverted to their pre-#255 state.

### CODE REVIEW

File: framework/mesh/mesh.h [hunk 1]
```diff
@@ -61,6 +61,16 @@ enum MeshAttributes : int
   UNSTRUCTURED = (1 << 5)
```
Comment (ragusa) [inline framework/mesh/mesh.h:61]: @wdhawkins @andrsd @zhardy-lanl  : I thought we had a discussion about bitwise operator for mesh enum and that we wanted to get rid of them (but we would keep the bitwise ops for source specifications). I just saw this here by chance. Should be create an issue?

**Review by ragusa — APPROVED** [review]

File: framework/mesh/mesh.h (hunk 1 above)
Comment (wdhawkins) [reply framework/mesh/mesh.h:61]: Done. See #286.

**Review by zhardy-lanl — APPROVED** [review]
Thank you for tracking this down! So am I correct in assuming that all boundary conditions were backwards? I don't know how I missed that on #255, if so. The boundary ID changes should not have touched the input language at any point. It should have been assigning different IDs to boundary names under the hood only.

**Review by andrsd — APPROVED** [review]

**Comment by wdhawkins** [comment]
> Thank you for tracking this down! So am I correct in assuming that all boundary conditions were backwards? I don't know how I missed that on #255, if so. The boundary ID changes should not have touched the input language at any point. It should have been assigning different IDs to boundary names under the hood only.

Yes, that's correct. Everything was backwards. I dug through the code pretty carefully, so I think I got everything this time.  🤞

**Pull request merged**